lib/cloudware/aws.rb

```diff
@@ -148,6 +149,36 @@ def machines
                     end
     end
 
+    def machine_info(name, domain)
+      @machine_info = {}
+      regions.each do |r|
+        load_config(r)
+        @ec2.describe_instances(filters: [
+                                  { name: 'tag:cloudware_domain', values: [domain.to_s] },
+                                  { name: 'tag:cloudware_machine_name', values: [name.to_s] }
+                                ]).reservations.each do |reservation|
+          reservation.instances.each do |instance|
+            @instance_id = instance.instance_id
+            @state = instance.state.name
+            @region = r
+          end
```

Comment:
This method's pretty dense - I'd probably split it up a bit more and in particular avoid the double `each` with a lot going on in between.